summarize/src/analysis.rs

```diff
@@ -20,141 +74,76 @@ pub fn perform_analysis(data: ProfilingData) -> Results {
+                }
+
+                // If there is something on the stack, subtract the current
+                // interval from it.
+                if let Some(current_top) = thread_stack.last() {
+                    record_event_data(&current_top.label, &|data| {
+                        data.self_time -= current_event.duration().unwrap();
                     });
+                }
 
-                    //this is the critical bit to correctly calculating self-time:
-                    //adjust the previous event's start time so that it "started" right now
-                    if let Some(previous_event) = thread_stack.last_mut() {
-                        assert_eq!(TimestampKind::Start, previous_event.timestamp_kind);
-                        previous_event.timestamp = event.timestamp;
+                // Update counters for the current event
+                match &current_event.event_kind[..] {
+                    QUERY_EVENT_KIND | GENERIC_ACTIVITY_EVENT_KIND => {
+                        record_event_data(&current_event.label, &|data| {
+                            data.self_time += current_event.duration().unwrap();
+                            data.number_of_cache_misses += 1;
+                            data.invocation_count += 1;
+                        });
                     }
 
-                    //record the total time
-                    total_time += duration;
-                } else if &event.event_kind[..] == QUERY_BLOCKED_EVENT_KIND {
-                    record_event_data(&event.label, &|data| {
-                        data.blocked_time += duration;
-                    });
-                } else if &event.event_kind[..] == INCREMENTAL_LOAD_RESULT_EVENT_KIND {
-                    record_event_data(&event.label, &|data| {
-                        data.incremental_load_time += duration;
-                    });
+                    QUERY_BLOCKED_EVENT_KIND => {
+                        record_event_data(&current_event.label, &|data| {
+                            data.blocked_time += current_event.duration().unwrap();
+                        });
+                    }
+
+                    INCREMENTAL_LOAD_RESULT_EVENT_KIND => {
+                        record_event_data(&current_event.label, &|data| {
+                            data.incremental_load_time += current_event.duration().unwrap();
+                        });
+                    }
+
+                    unknown_event_kind => {
+                        eprintln!(
+                            "Ignoring event with unknown event kind `{}`",
+                            unknown_event_kind
+                        );
+                    }
+                };
+
+                // Update the total_time counter
+                let current_event_is_top_level = thread_stack.is_empty();
+                if current_event_is_top_level {
+                    total_time += current_event.duration().unwrap();
```

Comment:
think that this is missing alot of time hade prefered if the first and last time for the thread/or the whole log was used

Reply:
in crox files I can see multiple seconds before and after analysis that is not covered by any event maybe there shall be an event started in the main function that cover everything

Reply:
Yeah, it's probably a good idea to measure both, the sum of top-level event times, and the sum of min/max times of all threads. Plus the wall-time.

In the last working group meeting we mentioned that we want to add a generic key-value map to the summarize output. The different "total time" measurements (and an estimate of time not accounted for) could go in there.

Reply:
I switched to the method you proposed, @andjo403.